specs/Standards/ContractMetadata.md

```diff
@@ -0,0 +1,49 @@
+# Contract Metadata
+
+## [NEP-330](https://github.com/near/NEPs/discussions/329)
+
+Version `1.0.0`
+
+## Summary
+
+The contract metadata is a standard interface to allow for auditing and viewing source code for a deployed smart contract. Implementation of this standard is purely optional but is recommended for developers whose contracts are open source.
+
+## Motivation
+
+There is no trivial way of finding the source code or author of a deployed smart contract. By having a standard that outlines how to view the source code of an arbitrary smart contract, an environment of openness and collaboration is created.
+
+The initial discussion can be found [here](https://github.com/near/NEPs/discussions/329).
+
+## Interface
+
+Metadata applies at the contract level (`ContractMetadata`):
+
+```ts
+type ContractMetadata = {
+  version: string|null, // optional, commit hash being used for the currently deployed wasm. If the contract is not open-sourced, this could also be a numbering system for internal organization / tracking such as "1.0.0" and "2.1.0".
+  link: string|null, //optional,  link to open source code such as a Github repository or a CID to somewhere on IPFS.
+}
+```
+
+A new function for querying the metadata must be supported on each smart contract implementing the standard:
+
+```ts
+function contract_metadata(): ContractMetadata {}
+```
+
+A new function for setting the contract metadata in case of updates to the currently deployed wasm or link to the open source code may optionally be implemented:
+
+```ts
+function set_contract_metadata(
+  metadata: ContractMetadata,
+): ContractMetadata {}
```

Comment:
why do you need this? wouldn't you just statically update what is returned when you upgrade the contract code? Or is this to be able to update the link when you don't want to upgrade the contract?

Reply:
That's a good point - you could just have everything as static variables in the contract but I kept it open in case people wanted to bump versions without having to redeploy the contract (for organizational purposes).

Reply:
but then if it's optional, maybe it shouldn't be a part of the standard? Kinda like mint for nfts?

Reply:
That's a good point - I'll change it and leave it up to the author to figure out they want to change the metadata.